src/system/SystemLayerImplSelect.cpp

```diff
@@ -197,8 +242,12 @@ CHIP_ERROR LayerImplSelect::ScheduleWork(TimerCompleteCallback onComplete, void
         });
         return CHIP_NO_ERROR;
     }
-#endif // CHIP_SYSTEM_CONFIG_USE_DISPATCH
-
+#elif CHIP_SYSTEM_CONFIG_USE_LIBEV
+    // just a timer with no delay
+    return StartTimer(Clock::Timeout(0), onComplete, appState);
```

Comment:
This will be buggy if someone calls ScheduleWork twice with the same args: it should schedule twice, but this will cancel the previous one, right?

The Darwin impl gets this right.  The select-based impl will get it right once https://github.com/project-chip/connectedhomeip/pull/22375 merges.

Reply:
> There's a not-entirely-obvious dependency here where ScheduleLambda on the System::Layer will run it via PostEvent on the platform manager...
> And of course various things directly PostEvent and whatnot.

It's beginning to dawn on me that this is more complicated - I wasn't aware that PostEvent also runs on dispatch, directly.

> Mostly I feel that event loops and the stuff around them are complex enough people get confused easily, so I would rather not have code around that makes them more confused.

I guess you're right. I'll remove the things related to that Darwin/libev Frankenstein's monster type of setup from the PR ;-)

After all, the libev support does *not* aim at replacing dispatch when it *is* available, but is useful for embedded linux targets like openwrt. Running the hybrid setup on the mac is just convenient for my develop-in-Xcode workflow, I can easily keep a few private patches for that.

> This will be buggy if someone calls ScheduleWork twice with the same args: it should schedule twice, but this will cancel the previous one, right?

Right. I'll have a look at #22375 and find a way to make the libev variant working in the same way.